rest/data/host_create.go

```diff
@@ -390,9 +390,15 @@ func makeEC2IntentHost(taskID, userID, publicKey string, createHost apimodels.Cr
 		ec2Settings.UserData = createHost.UserdataCommand
 	}
 
+	defaultSg := evergreen.GetEnvironment().Settings().Providers.AWS.DefaultSecurityGroup
+
 	// Always override distro security group with provided security group.
 	if len(createHost.SecurityGroups) > 0 {
 		ec2Settings.SecurityGroupIDs = createHost.SecurityGroups
+	} else if ec2Settings.SecurityGroupIDs == nil || len(ec2Settings.SecurityGroupIDs) == 0 {
```

Comment:
You don't need to have an else if and an else here; append handles the nil case correctly.
https://play.golang.org/p/Kcn-I4Sgb2D